reactioncommerce/reaction
# GraphQL: update fulfillment options for group

Resolves #4512  
Impact: **breaking**  
Type: **feature**

## Changes
- `updateFulfillmentOptionsForGroup` mutation now works
- `registerFunction` is now available on the context passed to a plugin startup function. `registerFunction("getShippingPrices", getShippingPrices);` is now the way to register a function for getting shipping prices (quotes).

## Breaking changes
- When getting shipping quotes, "onGetShippingRates" hooks are no longer used. Instead plugins should call `registerFunction("getShippingPrices", getShippingPrices);` following the pattern of the included shipping rates plugin. The provided function must not depend on Meteor or cause any Meteor packages to be imported. The arguments that are provided to this function are slightly different, too. They are now `(context, cart, previousQueryResults)`. Use `context.collections` to get collections for any database calls. (This pattern is subject to change.)
- The `"shipping/updateShipmentQuotes"` Meteor method now passes through to the `updateFulfillmentOptionsForGroup` mutation, and it takes two additional arguments (`fulfillmentGroupId, cartToken`). It is now called in fewer places and does not do as many adjustments of the fulfillment group object, instead assuming that they will be in the necessary state (exist, have a shipping address, have items, etc.) prior to the method being called.

## Testing
- Verify the mutation updates the shipping quotes on the specified `cart` fulfillment group.
- Verify that adding items to cart and checking out still works in Reaction Meteor app.
